delta/kernel/src/test/scala/ch/epfl/bluebrain/nexus/delta/kernel/utils/ClasspathResourceLoaderSpec.scala

```diff
@@ -1,66 +1,65 @@
-    }
-
-  "A ClasspathResourceUtils" should {
-    val resourceIO = loader.contentOf("resource.txt", "value" -> "v")
+  test("return the path") {
+    loader.absolutePath("resource.txt").assert(_.endsWith("resource.txt"))
+  }
 
-    "return the path" in {
-      accept(loader.absolutePath("resource.txt")) should endWith("resource.txt")
-    }
+  test("return the contents of a handlebar template") {
+    loader
+      .contentOf("resource.txt", "value" -> "v")
+      .assertEquals("A text resource with replacement 'v'")
+  }
 
-    "return a text" in {
-      accept(resourceIO) shouldEqual "A text resource with replacement 'v'"
-    }
+  test("return the contents of a handlebar template, multiple times") {
+    val io = loader.contentOf("resource.txt", "value" -> "v")
 
-    "read a second time" in {
-      accept(resourceIO) shouldEqual "A text resource with replacement 'v'"
+    for {
+      _ <- io.assertEquals("A text resource with replacement 'v'")
+      _ <- io.assertEquals("A text resource with replacement 'v'")
+    } yield {
+      ()
     }
+  }
 
-    "return a json" in {
-      accept(loader.jsonContentOf("resource.json", "value" -> "v")) shouldEqual Json.obj("k" -> "v".asJson)
-    }
+  test("return the contents of a handlebar template as json") {
+    loader
+      .jsonContentOf("resource.json", "value" -> "v")
+      .assertEquals(Json.obj("k" -> "v".asJson))
+  }
 
-    "return a json object" in {
-      accept(loader.jsonObjectContentOf("resource.json", "value" -> "v")) shouldEqual JsonObject("k" -> "v".asJson)
-    }
+  test("fail when a file cannot be parsed as json") {
+    loader
+      .jsonContentOf("resource.txt", "value" -> "v")
+      .intercept[InvalidJson]
+  }
 
-    "fail when resource is not a json" in {
-      reject(loader.jsonContentOf("resource.txt")) shouldBe a[InvalidJson]
-    }
+  test("return the contaents of a handlebar template as a json object") {
+    loader
+      .jsonObjectContentOf("resource.json", "value" -> "v")
+      .assertEquals(JsonObject("k" -> "v".asJson))
+  }
 
-    "fail when resource is not a json object" in {
-      reject(loader.jsonObjectContentOf("resource-json-array.json")) shouldEqual InvalidJsonObject(
-        "resource-json-array.json"
+  test("fail when a file contains JSON but is not a json object") {
+    loader
+      .jsonObjectContentOf("resource-json-array.json")
+      .intercept[InvalidJsonObject]
+      .assertEquals(
+        InvalidJsonObject("resource-json-array.json")
       )
-    }
+  }
 
-    "fail when resource does not exists" in {
-      reject(loader.contentOf("resource2.txt", "value" -> "v")).getMessage should (include("not found") and include(
-        "resource2.txt"
-      ))
-    }
+  test("fail when a resource does not exist") {
+    loader
+      .contentOf("resource2.txt", "value" -> "v")
+      .intercept[Throwable]
+      .assert(e => e.getMessage.contains("not found") && e.getMessage.contains("resource2.txt"))
```

Comment:
Not a new change and I'm not sure about asserting on library errors anyway, but if we are gonna do that could do something like

```suggestion
      .interceptMessage[IOException]("Resource resource2.txt not found")
```

Reply:
I wrote it this way because the message changed slightly when I switched libraries and I wanted the assertion to be a bit looser (say that it wasn't found, say the name of the thing)